functions.py

```diff
@@ -269,3 +270,234 @@
+    :type include_proportion: bool
+    :type include_within: bool
+    :type include_contains: bool
+    :param count:
+    :type count: int
+    :param offset:
+    :type offset: int
+    :return:
+    """
+    overlaps_sparql = """\
+PREFIX geo: <http://www.opengis.net/ont/geosparql#>
+PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
+PREFIX geox: <http://linked.data.gov.au/def/geox#>
+PREFIX qb4st: <http://www.w3.org/ns/qb4st/>
+PREFIX epsg: <http://www.opengis.net/def/crs/EPSG/0/>
+PREFIX dt: <http://linked.data.gov.au/def/datatype/>
+SELECT <SELECTS>
+WHERE {
+    {
+        {
+           ?s1 rdf:subject <URI> ;
+           rdf:predicate geox:transitiveSfOverlap;
+           rdf:object ?o  .
+        } UNION {
+           ?s2 rdf:subject <URI> ;
+           rdf:predicate geo:sfOverlaps;
+           rdf:object ?o  .
+        }
+    }
+    UNION
+    { <URI> geox:transitiveSfOverlap ?o }
+    UNION
+    { <URI> geo:sfOverlaps ?o }
+    <EXTRAS>
+}
+GROUP BY ?o
+"""
+    contains_sparql = """\
+    PREFIX geo: <http://www.opengis.net/ont/geosparql#>
+    PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
+    PREFIX geox: <http://linked.data.gov.au/def/geox#>
+    PREFIX qb4st: <http://www.w3.org/ns/qb4st/>
+    PREFIX epsg: <http://www.opengis.net/def/crs/EPSG/0/>
+    PREFIX dt: <http://linked.data.gov.au/def/datatype/>
+    SELECT ?c <SELECTS>
+    WHERE {
+        {  
+            ?s2 rdf:subject <URI> ;
+            rdf:predicate geo:sfContains;
+            rdf:object ?o  .
+        }
+        UNION
+        { <URI> geo:sfContains+ ?o }
+        OPTIONAL { FILTER(bound(?o))
+            BIND(true as ?c) .
+        }
+        <EXTRAS>
+    }
+    GROUP BY ?c ?o
+    """
+    within_sparql = """\
+    PREFIX geo: <http://www.opengis.net/ont/geosparql#>
+    PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
+    PREFIX geox: <http://linked.data.gov.au/def/geox#>
+    PREFIX qb4st: <http://www.w3.org/ns/qb4st/>
+    PREFIX epsg: <http://www.opengis.net/def/crs/EPSG/0/>
+    PREFIX dt: <http://linked.data.gov.au/def/datatype/>
+    SELECT ?w <SELECTS>
+    WHERE {
+        {  
+            ?s2 rdf:subject <URI> ;
+            rdf:predicate geo:sfWithin;
+            rdf:object ?o  .
+        }
+        UNION
+        { <URI> geo:sfWithin+ ?o }
+        OPTIONAL { FILTER(bound(?o))
+            BIND(true as ?w) .
+        }
+        <EXTRAS>
+    }
+    GROUP BY ?w ?o
+    """
+    use_areas_sparql = include_proportion or include_areas
+    use_proportion_sparql = include_proportion
+
+    selects = "?o "
+    area_selects = "(MAX(?a1) as ?uarea) (MAX(?a2) as ?oarea) "
+    iarea_selects = "(MAX(?a3) as ?iarea) "
+
+    areas_sparql = """\
+    OPTIONAL {
+        <URI> geox:hasAreaM2 ?ha1 .
+        ?ha1 qb4st:crs epsg:3577 .
+        ?ha1 dt:value ?a1 .
+    }
+    OPTIONAL {
+        ?o geox:hasAreaM2 ?ha2 .
+        ?ha2 qb4st:crs epsg:3577 .
+        ?ha2 dt:value ?a2 .
+    }
+    """
+    iarea_sparql = """\
```

Comment:
Eventually we should be able to get rid of this bit of the SPARQL especially if we store overlap proportion directly on sfOverlap reified statements. This should speed things up a bit

Reply:
Yes thats right, that part can be swapped out with a different chunk that reads the proportion directly.